### DISCUSSION TITLE
Allow ```default``` to infer from usage

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#3969] [Language Ideas]
This currently gives the error: "There is no target type..." but I think it should infer the type from the other arms of the coalesce.

```
                var BusinessAddress = default
                    ?? (from x in Addresses where x.Dest_Location == ImportAddressLocation.Billing select x).FirstOrDefault()
                    ?? (from x in Addresses where x.Dest_Location == ImportAddressLocation.Work select x).FirstOrDefault()
                    ;

```

**Comment by YairHalberstadt** [comment]
Isn't `default ?? expr` just always `expr`?

**Reply by TonyValenti** [reply]
Yes, but I like having it there for formatting purposes.

**Reply by YairHalberstadt** [reply]
I think it unlikely the team will add a new feature which is only useful for formatting purposes. Type inference can actually be quite complex and can change things like overload resolution in subtle ways.

**Reply by YairHalberstadt** [reply]
Also you could just do `null`
